### ISSUE TITLE
Pyspark code unable to discard bad records

### ISSUE THREAD

**Issue by Meethun** [Meethun/spark-csv#1]
Hey Gang - I am facing challanges in one of the code that i am writing into. My dataset has some weird record as mentioned below and i want to discard those records.

3762|696|100004555|100427914|Blazin' Rewards|blazinrewards@emailinfo.abc.com|2019-03-28 13:42:00|2019-03-28 13:43:00|2019-03-28 13:43:00||False|MULTIPLE_SEND|New|0|2019-03-28 13:42:35.963000000|BNS535|*PROD - BNS535* SHHH! You get points just for drinking beer                                                                                                                                             |True||my tracking|||2019-03-28 13:42:00|Unicode - UTF-8||0|0|ExactTargetSend||False|Default Commercial|Default Commercial|False|10158|False||
3763|708|100004555|100427914|Blazin' Rewards|blazinrewards@emailinfo.abc.com|2019-03-28 13:43:00|2019-03-28 13:44:00|2019-03-28 13:44:00||False|MULTIPLE_SEND|New|0|2019-03-28 13:43:35.980000000|DIP100409|*PROD - 	
DIP100409* 🏀🏀🏀 is life!!                                                                                                                                                                  |True||my tracking|||2019-03-28 13:43:00|Unicode - UTF-8||0|0|ExactTargetSend||False|Default Commercial|Default Commercial|False|10158|False||
3764|711|100004555|100427914|Blazin' Rewards|blazinrewards@emailinfo.abc.com|2019-03-28 13:43:00|2019-03-28 13:44:00|2019-03-28 13:44:00||False|MULTIPLE_SEND|New|0|2019-03-28 13:43:36.120000000|DIP100409 - 20190328_115500|*PROD - 	
DIP100409*  🏀🏀🏀 is life!!                                                                                                                                                                 |True||my tracking|||2019-03-28 13:43:00|Unicode - UTF-8||0|0|ExactTargetSend||False|Default Commercial|Default Commercial|False|10158|False||
3765|712|100004555|100427914|Blazin' Rewards|blazinrewards@emailinfo.abc.com|2019-03-28 13:44:00|2019-03-28 13:45:00|2019-03-28 13:45:00||False|MULTIPLE_SEND|New|0|2019-03-28 13:44:36.263000000|DIP100409 - 20190328_115533|*PROD - 	
DIP100409* 🏀🏀🏀 is life!!                                                                                                                                                                  |True||my tracking|||2019-03-28 13:44:00|Unicode - UTF-8||0|0|ExactTargetSend||False|Default Commercial|Default Commercial|False|10158|False||
3766|713|100004555|100427914|Blazin' Rewards|blazinrewards@emailinfo.abc.com|2019-03-28 13:44:00|2019-03-28 13:45:00|2019-03-28 13:45:00||False|MULTIPLE_SEND|New|0|2019-03-28 13:44:36.577000000|DIP100409 - 20190328_115550|*PROD - 	DIP100409* 🏀🏀🏀 is life!!                                                                                                                                                                  |True||my tracking|||2019-03-28 13:44:00|Unicode - UTF-8||0|0|ExactTargetSend||False|Default Commercial|Default Commercial|False|10158|False||
3767|700|100004555|100427914|Blazin' Rewards|blazinrewards@emailinfo.abc.com|2019-03-28 13:45:00|2019-03-28 13:46:00|2019-03-28 13:46:00||False|MULTIPLE_SEND|New|0|2019-03-28 13:45:36.220000000|PTS100403|*PROD - PTS100403* Hey %%FIRSTNAME%%, you're welcome.                                                                                                                                                   |True||my tracking|||2019-03-28 13:45:00|Unicode - UTF-8||0|0|ExactTargetSend||False|Default Commercial|Default Commercial|False|10158|False||

**Comment by aravish** [comment]
![image](https://user-images.githubusercontent.com/43964587/63973313-22c51500-ca70-11e9-9389-034c3243405c.png)

**Comment by aravish** [comment]
How to ignore records 67,69,71,73 when reading CSV in pyspark?

**Comment by Meethun** [comment]
yeah, i want to get those records having bad data removed or substitute with any special character such as * or simply "a space". Please note that such weird data can happen in any column and may not start with DIP******

**Comment by aravish** [comment]
Can we get an update please?

**Comment by Meethun** [comment]
Still a problem.. however, if I do not use MALFORMATTED, it is able to discard 4 records.
If I use malformatted, it was discarding 24 records.

Meethun Panda
Lead Data Engineer

M: +1 646 675 0594
BCG PLATINION

**Comment by aravish** [comment]
If I do not use MALFORMED, it is able to discard 4 records. Does it mean this solution works for you?

**Comment by Meethun** [comment]
Your solution was to use MULTILINE, which seems to be buggy.

I’ll recheck tomorrow and will let you know.

Meethun Panda
Lead Data Engineer

M: +1 646 675 0594
BCG PLATINION
